### PR TITLE
Removing AppEngine restriction on Java version.

### PR DESCRIPTION
**Pull request by Vizerai**
AppEngine uses Java 8 now.

### CODE REVIEW

File: impl_core/src/main/java/io/opencensus/implcore/internal/DaemonThreadFactory.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
   // that need to be accommodated.
   public static final boolean IS_RESTRICTED_APPENGINE =
       System.getProperty("com.google.appengine.runtime.environment") != null
-          && "1.7".equals(System.getProperty("java.specification.version"));
+          && "1.8".equals(System.getProperty("java.specification.version"));
```
Comment (bogdandrutu) [inline impl_core/src/main/java/io/opencensus/implcore/internal/DaemonThreadFactory.java]: I think for Java 8 we don't need to do this. In the Java 8 appengine they removed this requirement. Can you please check that?

Comment (Vizerai) [reply impl_core/src/main/java/io/opencensus/implcore/internal/DaemonThreadFactory.java]: It looks like AppEngine still does not support renaming of threads.  thread.setName() isn't allowed.

Comment (bogdandrutu) [reply impl_core/src/main/java/io/opencensus/implcore/internal/DaemonThreadFactory.java]: ok if that is the case do this:

```
  @Override
  public Thread newThread(Runnable r) {
    Thread thread = threadFactory.newThread(r);
    try {
      thread.setName(name);
      thread.setDaemon(true);
    } catch (SecurityException e) {
      // OK if we can't set the name or daemon in this environment.
    }
    return thread;
  }
```

This is also how guava implements this in `MoreExecutors.newThread()` which unfortunately is not public.

This way we remove the IS_RESTRICTED_APPENGINE and we are ok. What do you think?

Comment (Vizerai) [reply impl_core/src/main/java/io/opencensus/implcore/internal/DaemonThreadFactory.java]: Yes, we can do that.

**Comment by bogdandrutu** [comment]
Please update the description and name of the PR.

**Comment by Vizerai** [comment]
AppEngine restriction on Java version has been removed and replaced with an SecurityException check when trying to rename the thread or set it as a daemon thread.

**Comment by rghetia** [comment]
thanks @songy23 . This issue doesn't fix #1954.

**Pull request merged**